### DISCUSSION TITLE
Installation of Peridigm on HPC Cluster

### DISCUSSION THREAD

**Discussion by sarthaks20** [peridigm/peridigm#303] [General]
Hello!

Can anyone please provide a complete tutorial on how to setup Peridigm along with all of its dependencies on HPC Cluster as a user locally (who does not have sudo or root rights)?

Thank you.

**Comment by oldninja** [comment]
[PeriDoX](https://github.com/PeriDoX/PeriDoX/tree/master/Doc/Peridigm_Installation_Guide)

**Reply by sarthaks20** [reply]
I tried that but still Peridigm is not working.

**Reply by oldninja** [reply]
That installation guide is your best bet. However, I recommend starting from scratch and following step by step, it is very comprehensive, and I can tell you that I have used it to install it on an HPC cluster to which I did not have root access.

**Reply by sarthaks20** [reply]
@oldninja can you please tell the version of Trilinos you installed?

**Reply by oldninja** [reply]
@sarthak-iitd23 I have installed several versions, but trilinos-12.4.2 was a good start.

**Reply by sarthaks20** [reply]
@oldninja Should I install the latest version of Trilinos?

**Reply by oldninja** [reply]
I don't see why not. Just be prepared for debugging as needed.

**Reply by sarthaks20** [reply]
@oldninja I mean to ask if you faced any compatibility issue between latest version of Trilinos and Peridigm.

By the way, thanks a lot for your quick response. :)

**Reply by oldninja** [reply]
I have not tried the latest trilinos, so I don't know. I always try to go for the versions that others have tried to avoid compatibility issues. Install the one from the user guide (peridox) and go on from there. Once you have built it a couple of times, you will get a good feeling and try updated versions.

**Reply by sarthaks20** [reply]
Thanks!

**Comment by djlittl** [comment]
Regarding the issue of not having root access, that should not be a problem.  Just install everything in your user space.  All the packages have an option to specify the install directory.  You just need to pick one that you have write access to.

**Comment by sarthaks20** [comment]
While building Peridigm, I am facing the following error:


**Trilinos installation:
   Trilinos_DIR = /scratch/civil/btech/ce119xxx/Packages/trilinos/lib/cmake/Trilinos
   Trilinos_VERSION = 13.5
   Trilinos_PACKAGE_LIST = Piro;ROL;Stokhos;Tempus;Rythmos;MueLu;NOX;Phalanx;Intrepid2;Intrepid;Teko;Stratimikos;Ifpack2;Anasazi;SEACAS;SEACASEx2ex1v2;SEACASTxtexo;SEACASSlice;SEACASNumbers;SEACASNemspread;SEACASNemslice;SEACASZellij;SEACASNas2exo;SEACASMat2exo;SEACASMapvar-kd;SEACASMapvar;SEACASMapvarlib;SEACASExplore;SEACASGrepos;SEACASGenshell;SEACASGen3D;SEACASGjoin;SEACASFastq;SEACASEx1ex2v2;SEACASExo_format;SEACASExotxt;SEACASExomatlab;SEACASExodiff;SEACASExo2mat;SEACASCpup;SEACASEpu;SEACASEjoin;SEACASConjoin;SEACASBlot;SEACASAprepro;SEACASAlgebra;SEACASPLT;SEACASSVDI;SEACASSuplibCpp;SEACASSuplibC;SEACASSuplib;SEACASSupes;SEACASAprepro_lib;SEACASChaco;SEACASIoss;SEACASNemesis;SEACASExoIIv2for32;SEACASExodus_for;SEACASExodus;Amesos2;ShyLU_Node;ShyLU_NodeTacho;ShyLU_NodeHTS;Belos;ML;Ifpack;Zoltan2Core;Pamgen;Amesos;Galeri;AztecOO;Isorropia;Xpetra;Thyra;ThyraTpetraAdapters;ThyraEpetraExtAdapters;ThyraEpetraAdapters;ThyraCore;TrilinosSS;Tpetra;TpetraCore;TpetraTSQR;EpetraExt;Triutils;Shards;Zoltan;Epetra;MiniTensor;Sacado;RTOp;KokkosKernels;Teuchos;TeuchosKokkosComm;TeuchosKokkosCompat;TeuchosRemainder;TeuchosNumerics;TeuchosComm;TeuchosParameterList;TeuchosParser;TeuchosCore;Kokkos;KokkosSimd;KokkosAlgorithms;KokkosContainers;KokkosCore;Gtest
   Trilinos_TPL_LIST =
   Trilinos_TPL_LIBRARY_DIRS =
   Trilinos_BUILD_SHARED_LIBS = FALSE
   Trilinos_CXX_COMPILER_FLAGS = -O2 -std=c++11 -pedantic -ftrapv -Wall -Wno-long-long   -O3 -DNDEBUG
   Trilinos_INCLUDE_DIRS = /scratch/civil/btech/ce119xxx/Packages/trilinos/include
   Trilinos_LIBRARY_DIRS =

FATAL_ERROR

Error:  Netcdf NOT found.  Netcdf is a REQUIRED Trilinos TPL.**



I have installed NetCDF 4.9.0, but still getting this error.
